### DISCUSSION TITLE
Why is no Service created for the metrics port?

### DISCUSSION THREAD

**Discussion by UntouchedWagons** [kubernetes-sigs/node-feature-discovery#1593] [Q&A] [answered]
In the helm chart there's a metrics port but no Service gets created for it, why is that?

**Comment by marquiz** [comment, marked as answer]
Hi @UntouchedWagons, a *bit* delayed answer, but I think the answer is that it was not needed when that part was written. If prometheus monitoring is enabled via the helm chart we monitor pods directly (with the `PodMonitor` objects) and no services are needed. Maybe @ArangoGutierrez can comment, too, as the original author of the metrics support for NFD.

@UntouchedWagons would you have need for a service?

**Reply by ArangoGutierrez** [reply]
+1 for `PodMonitor` , that is right

**Comment by UntouchedWagons** [comment]
No no that's okay I was just kinda confused. My understanding was that if pod A wants to interact with pod B somehow it has to be through a Service object. It looks like I was mistaken.
